### PR TITLE
Insert favorites when bookmark is favorite and no index

### PR DESCRIPTION
**Pull request by jotaemepereira**
Task/Issue URL: https://app.asana.com/0/1201048563534612/1208595122638622/f
Tech Design URL:
CC:

**Description**:
It fixes new bookmarks not being added to favorites from the dialog when no favorites index is passed.

**Steps to test this PR**:
1. Create new bookmark from the dialog or by going to the new tab page and tapping the add favorite cell
2. Check the ‘Add as favorite’ box
3. Add the bookmark
4. The bookmark should be part of the new tab page array of favorites (you should also check is a favorite in the panel and in the bookmarks manager)

**Definition of Done**:

* [x] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

**Review by ayoy — COMMENTED** [review]
The testing went great @jotaemepereira, thanks for this fix 👏  I've checked inserting favorites, deleting and reordering, and also various Sync scenarios and haven't noticed any inconsistencies.

I'd appreciate if we could add unit tests for this.

File: DuckDuckGo/Bookmarks/Services/LocalBookmarkStore.swift [hunk 1]
```diff
@@ -322,6 +322,7 @@ final class LocalBookmarkStore: BookmarkStore {
             }
 
             try restoreFavorites(entitiesAtIndices, entitiesByUUID: entitiesByUUID, in: context)
+            try insertNewFavorites(entitiesAtIndices, entitiesByUUID: entitiesByUUID, in: context)
```
[the remark this answers is not in the document]
Comment (ayoy) [reply DuckDuckGo/Bookmarks/Services/LocalBookmarkStore.swift:325]: `sortEntitiesByFavoritesIndex` filters out nil indices so in current shape it won't work at all - this is the cause of the bug. We could update it to put nil indices at the end but I feel like it's too much unnecessary logic for just appending favorites.

[the remark this answers is not in the document]
Comment (jotaemepereira) [reply DuckDuckGo/Bookmarks/Services/LocalBookmarkStore.swift]: @ayoy @mallexxx I’ve refactored the code a bit and fixed the unit tests (we already had unit tests, the problem was that I made a mistake when fixing them where I added the index manually when it should have been nil).

About the code duplication, I want to keep two separate functions: one for restoring and another one for inserting; it makes the code more readable and easiere to follow.

**Review by ayoy — APPROVED** [review]
LGTM, thanks a lot @jotaemepereira !

**Pull request merged**